docs/src/pages/ThemingSpecification.mdx

```diff
@@ -0,0 +1,185 @@
+export const meta = {
+  title: 'Theming specification',
+}
+
+## Goals and requirements
+
+The theming mechanism as part of Fluent UI should be flexible and provide support for as many scenarios as possible
+
+##### Support for CSS stylesheets
+
+Тhe users should be able to specify mappings from properties to classnames
+
+**IMPORTANT**: applications which load their own CSS in other means should not pay the penalty of loading a runtime CSS-in-JS package, especially if the class names are known at time of authoring.
+
+##### Support for CSSinJS
+
+CSSinJS support that will not tie the users to just one CSS in JS framework. The users should be able to define their styles and specify their engine that will be aligned to some defined interface.
+
+#### A component should be able to render any theme (including tokens, styles, icons and animations)
+
+The client should be able to reuse the components between different themes. Supporting new theme in the application should not require building new components.
+
+If new variants are added in the new theme, then the clients should be able to compose the existing component, by adding new design terms (via the props API). If the new theme doesn't need to support any new variant, nothing should be required from the client's side.
+
+##### Variants
+
+Variants are a way of allowing the users to specify new design terms for their components (among other customizations).
+
+In the CSS scenario, it is basically a way of allowing user to specify mapping to className for a new prop.
+
+In the CSSinJS scenario, it is allowing user to have new properties (design terms) inside the styles functions.
+
+#### Themes should cascade and merge allowing subtree themes to be updated at runtime and have the ability to adjust the incoming theme.
+
+If some parameters/tokens are changed on the root of the application, the changes should be propagated to all nested subtrees.
+
+#### RTL and disable animation modes should be configurable for the entire application or a subtree of the application.
+
+It is important to allow the users to specify the RTL/LTR mode for the entire application once, but if necessary they can specify it only in a subtree of the application (for example one ChatMessage)
+
+The same is true for the disable animations mode. It is important to allow the users to specify it for the entire application once, but if necessary they can specify it only in a subtree of the application (for example the `Loader` component should always be animated).
+
+#### Parent components should be able to customize the styles for the child components
+
+The slots prop must be accessible in several means to ensure that a "recursive component" can be accurately styled.
```

Comment:
Is a recursive component the only use-case? If so, it seems like there are several other ways to accomplish the same. (eg pass a prop to a child and have the child make decisions based on that prop.)

Reply:
Not really, below I have a details table with supporting parent styling slots/children. I will expand this to that scenario as well.

Reply:
I've updated this section. I understand that this can be achieved using different strategies, but we have the comparison in the table below anyway